### PR TITLE
Use cardano-cli create-testnet-data instead of create-staked

### PR DESCRIPTION
**Pull request by Jimbo4350**
# Description

Use `create-testnet-data` because `create-staked` was somewhat rushed into `cardano-cli`. We plan to improve and potentially make breaking changes to `create-testnet-data` and we want to avoid breaking existing functionality in `create-staked`.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/KesPeriodInfo.hs [hunk 1]
```diff
@@ -67,39 +65,32 @@ hprop_kes_period_info = H.integrationRetryWorkspace 2 "kes-period-info" $ \tempA
...
-      , "--cardano-mode"
-      , "--testnet-magic", show @Int testnetMagic
-      , "--out-file", work </> "utxo-1.json"
-      ]
-
-  H.cat $ work </> "utxo-1.json"
+  let utxoAddr = Text.unpack $ paymentKeyInfoAddr $ head wallets
+      utxoSKeyFile = paymentSKey . paymentKeyInfoPair $ head wallets
+  void $ H.execCli' execConfig
+    [ "babbage", "query", "utxo"
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/KesPeriodInfo.hs:78]: Update with a programmatic era

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/KesPeriodInfo.hs [hunk 2]
```diff
@@ -153,13 +144,13 @@ hprop_kes_period_info = H.integrationRetryWorkspace 2 "kes-period-info" $ \tempA
...
 
-  void $ execCli' execConfig
-      [ "query", "utxo"
-      , "--address", utxoAddr
-      , "--cardano-mode"
-      , "--testnet-magic", show @Int testnetMagic
-      , "--out-file", work </> "utxo-2.json"
-      ]
+  void $ H.execCli' execConfig
+    [ "babbage", "query", "utxo"
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/KesPeriodInfo.hs:148]: add programmatic era

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 3]
```diff
@@ -189,3 +210,15 @@ mkTopologyConfig numNodes allPorts port True = Aeson.encode topologyP2P
...
+convertToEraString :: AnyCardanoEra -> String
+convertToEraString (AnyCardanoEra e) =
+  case e of
+    ConwayEra -> "conway"
+    BabbageEra -> "babbage"
+    AlonzoEra -> "alonzo"
+    MaryEra -> "mary"
+    AllegraEra -> "allegra"
+    ShelleyEra -> "shelley"
+    ByronEra -> "byron"
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:224]: You can use this instead:
```suggestion
convertToEraString = map toLower . docToString . pshow
```

File: cardano-testnet/src/Testnet/Property/Utils.hs [hunk 4]
```diff
@@ -89,7 +89,7 @@ isLinux = os == "linux"
 
 -- | Submit the desired epoch to wait to.
 waitUntilEpoch
-  :: (MonadCatch m, MonadIO m, MonadTest m)
+  :: (MonadCatch m, MonadIO m, MonadTest m, HasCallStack)
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Property/Utils.hs:92]: Nice! `HasCallStack` is always useful.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs [hunk 5]
```diff
@@ -87,33 +86,22 @@ hprop_leadershipSchedule = H.integrationRetryWorkspace 2 "babbage-leadership-sch
...
+    , "--address", utxoAddr
+    , "--cardano-mode"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--out-file", work </> "utxo-1.json"
+    ]
 
   utxo1Json <- H.leftFailM . H.readJsonFile $ work </> "utxo-1.json"
-  UTxO utxo1 <- H.noteShowM $ decodeEraUTxO sbe utxo1Json
-  txin <- H.noteShow =<< H.headM (Map.keys utxo1)
+  UTxO utxo1 <- H.noteShowM $ H.noteShowM $ decodeEraUTxO sbe utxo1Json
```
Comment (carbolymer) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:100]: why duplicated `noteShowM`?

**Review by carbolymer — APPROVED** [review]

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs (hunk 5 above)
Comment (Jimbo4350) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/LeadershipSchedule.hs:100]: copy paste error, I'll fix it once the branch passes CI

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 6]
```diff
@@ -117,16 +123,24 @@ createSPOGenesisAndFiles testnetOptions startTime (TmpAbsolutePath tempAbsPath')
       . HM.insert "updateQuorum"           (toJSON @Int 2)
       )
 
+  -- TODO: create-testnet-data should have arguments for
+  -- Alonzo and Conway genesis that are optional and if not
+  -- supplised the users get a default
```
Comment (smelc) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:128]: FYI this is tracked here: https://github.com/IntersectMBO/cardano-cli/issues/548 so it'll happen at some point

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs [hunk 7]
```diff
@@ -48,7 +48,7 @@ newtype AdditionalCatcher
 
 
 hprop_ledger_events_propose_new_constitution :: Property
-hprop_ledger_events_propose_new_constitution = H.integrationRetryWorkspace 2 "propose-new-constitution" $ \tempAbsBasePath' -> do
+hprop_ledger_events_propose_new_constitution = H.integrationWorkspace "propose-new-constitution" $ \tempAbsBasePath' -> do
```
Comment (smelc) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs:51]: Given the CI's behavior the last days, this is bold :smile:

**Review by smelc — APPROVED** [review]
Nice :+1:

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs (hunk 7 above)
Comment (Jimbo4350) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs:51]: IIRC this was due to the timeouts in our tests. These properties don't rely on timeouts.

File: cardano-testnet/src/Testnet/Components/Configuration.hs (hunk 3 above)
Comment (Jimbo4350) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:224]: This actually gives "conwayera". I'm reverting to what I originally had.

**Comment by smelc** [comment]
I rebased the branch.

**Pull request merged**